src/reader_async.rs

```diff
@@ -103,20 +107,30 @@ trait SharedIO<T: AsyncFileReader + Unpin + Clone + 'static> {
 
 #[wasm_bindgen]
 pub struct AsyncParquetFile {
-    reader: HTTPFileReader,
+    reader: ParquetObjectReader,
     meta: ArrowReaderMetadata,
     batch_size: usize,
     projection_mask: Option<ProjectionMask>,
 }
 
 impl SharedIO<HTTPFileReader> for AsyncParquetFile {}
+impl SharedIO<ParquetObjectReader> for AsyncParquetFile {}
 
 #[wasm_bindgen]
 impl AsyncParquetFile {
     #[wasm_bindgen(constructor)]
-    pub async fn new(url: String) -> WasmResult<AsyncParquetFile> {
-        let client = Client::new();
-        let mut reader = HTTPFileReader::new(url.clone(), client.clone(), 1024);
+    pub async fn new(url: String, options: Option<Object>) -> WasmResult<AsyncParquetFile> {
+        let parsed_url = Url::parse(&url)?;
+        let (storage_container, path) = match options {
+            Some(options) => {
+                let deserialized_options: HashMap<String, String> =
+                    serde_wasm_bindgen::from_value(options.into())?;
+                parse_url_opts(&parsed_url, deserialized_options.iter())?
+            }
+            None => parse_url_opts(&parsed_url, std::iter::empty::<(String, String)>())?,
```

Comment:
Can we infer the types here? Or does `parse_url_opts` use a generic there?
```suggestion
            None => parse_url_opts(&parsed_url, std::iter::empty())?,
```

Reply:
Yep, it does - that being said, there's an option-free export that does this for us.